### DISCUSSION TITLE
Nach Gewitter (ohne Sonne) DTU verliert Tagesertrag und startet neu

### DISCUSSION THREAD

**Discussion by oldieman** [tbnobody/OpenDTU#1272] [General]
**Folgendes Scenario**: Sonne scheint und Tagesertrag wird angezeigt und aufsummiert. Jetzt kommt ein Gewitter und die Sonne verschwindet für eine halbe Stunde und kommt dann wieder. Der Tagesertrag sitzt jetzt auf 0 und startet neu.
Welche Abhilfe könnte die DTU machen? Sie hat die Uhrzeit und weiß, dass es noch nicht Nacht ist und noch Strom produziert werden könnte.

**Comment by jimjackii** [comment]
Das ist normal, nach Neustart beginnt der WR neu zu zählen. Das sollte kein Problem sein und tritt ja nicht alle Tage auf. Der Gesamtertrag gibt doch genug Auskunft.

**Reply by oldieman** [reply]
Ist doch ein Problem wenn man tägliche "**Buchführung**" über seine Solaranlagen (nicht nur Balkon) und Zählerstände führt.

**Comment by apreller64** [comment]
das passiert auch sehr häufig bei bewölktem Himmel früh morgens und Abends bei geringem Ertrag. Schiebt sich eine Wolke vor die
Sonne beginnt der WR eine neue Zählung.
Da die DTU keinerlei Werte speichert muss man sich anderwaltig helfen. Entweder täglich den Gesamtertrag in die "Buchführung" einfließen lassen und den Tagesertrag aus der Differenz errechnen oder die Werte in einer Datenbank speichern so wie das [hier](https://github.com/Kraego/OpenDTU-Grafana-Howto/blob/main/OpenDTUGrafana.md) beschrieben ist.

**Comment by turrican944** [comment]
Ich habe das problem mit Homeassistant gelöst in dem ich den Tagesertrag der WR in einen Zähler packe der mit daraus den Tagesetrag errechnent. Homeassistant macht dann bei seinem Zähler am 0 Uhr einen Reset.
Für mich reicht das völlig aus da ich OpenDTU nur als Gateway nutze (so ist das ja eigentlich auch gedacht). Es gibt aber einen Fork von OpenDTU mit Datenbank, evtl macht der das genau so wie du möchtest.

**Reply by LeChrisDE** [reply]
Für viele Interessenten würde es vermutlich auch hilfreich sein, wenn am Ende des Sonnentages (Leistung geht gegen 0) eine E-Mail an eine zu konfigurierende Adresse mit dem Tagesergebnis versandt werden könnte.
